src/sentry_core.c

```diff
@@ -689,3 +710,79 @@ sentry_set_level(sentry_level_t level)
         scope->level = level;
     }
 }
+
+sentry_value_t
+sentry_transaction_start(sentry_value_t tx_cxt)
+{
+    sentry_value_t tx = sentry_value_new_event();
+
+    // TODO(tracing): stuff transaction into the scope
+    bool should_sample = sentry__should_send_transaction(tx_cxt);
+    sentry_value_set_by_key(
+        tx, "sampled", sentry_value_new_bool(should_sample));
+
+    // Avoid having this show up in the payload at all if it doesn't have a
+    // valid value
+    sentry_value_t parent_span
+        = sentry_value_get_by_key_owned(tx_cxt, "parent_span_id");
+    if (sentry_value_get_length(parent_span) > 0) {
```

Comment:
rather minor as the semantics end up being the same, but i was kind of expecting a `!sentry_value_is_null(parent_span)` here

Reply:
yeah, i was really relying on the fact that `.._get_length()` handles nulls already here. a lot of the helper functions for `sentry_value_t` do so already, so i'm trying to remove any unnecessary null checks whereever i can as a result.